packages/components/psammead-content-shift-blocker/src/index.jsx

```diff
@@ -0,0 +1,159 @@
+import React, { useEffect, useLayoutEffect, useRef, useState } from 'react';
+import when from 'ramda/src/when';
+import is from 'ramda/src/is';
+import { number, string, node, oneOfType } from 'prop-types';
+import styled from 'styled-components';
+
+const convertToPixels = num => `${num}px`;
+
+const getSize = when(is(Number), convertToPixels);
+
+const getHeight = ({ wrapperHeight }) => getSize(wrapperHeight);
+
+const getWidth = ({ wrapperWidth }) => getSize(wrapperWidth);
+
+const Wrapper = styled.div`
+  overflow: auto;
+  height: ${getHeight};
+  width: ${getWidth};
+`;
+
+const initIntersectionObserver = ({ wrapperEl, setWrapperIO }) => {
+  const init = () => {
+    const callback = ([wrapperEntry]) => {
+      setWrapperIO(wrapperEntry);
+    };
+    const IO = new IntersectionObserver(callback);
+
+    IO.observe(wrapperEl.current);
+
+    return IO;
+  };
+
+  if ('IntersectionObserver' in window) {
+    return init();
+  }
+
+  return import('intersection-observer').then(() => {
+    IntersectionObserver.prototype.POLL_INTERVAL = 100;
+    return init();
+  });
+};
+
+const initResizeObserver = ({ wrapperEl, setContentElRect }) => {
+  const init = ResizeObserver => {
+    const callback = ([contentEntry]) => {
+      setContentElRect(contentEntry.contentRect);
+    };
+    const RO = new ResizeObserver(callback);
+
+    RO.observe(
+      wrapperEl.current
+        .firstChild /* will break if using a fragment :thinking: */,
```

Comment:
Would be good to explicitly mention in the Readme that this component does not support passing in a Fragment wrapping several children, but that it should be e.g. a `div`.